src/json.h

```diff
@@ -703,6 +706,71 @@ class JsonObject
 
         // useful debug info
         std::string line_number(); // for occasional use only
+
+        template <typename T>
```

Comment:
The long term goal is to actually have `assign` update an internal member of `JsonObject` so that we can detect unused fields. That's obviously very coupled to the JSON logic.